ehsm_kms_service/secret_manager_apis.js

```diff
@@ -151,8 +186,68 @@ function calculateRotationSperate(rotationInterval, createTime) {
     return nextRotationDate
 }
 
-//create Secret
-const createSecret = async (res, appid, payload, DB) => {
+/**
+ * Force delete secret metadata and secret version data by appId and secretName
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controller
+ * @param {Object} secretName : The name of the secret. eg. 'secretName01'
+ * @returns {boolean}
+ */
+const forceDeleteData = async (DB, appid, secretName) => {
+    try {
+        //Query and delete the secret metadata
+        const query_secret_metadata = {
+            selector: {
+                appid,
+                secretName
+            },
+            fields: ['_id', '_rev'],
+            limit: 1
+        }
+        let secret_metadata_res = await DB.partitionedFind('secret_metadata', query_secret_metadata)
+        if (secret_metadata_res.docs.length > 0) {
+            await DB.destroy(secret_metadata_res.docs[0]._id, secret_metadata_res.docs[0]._rev)
+            //Query and delete the secret version data
+            const query_secret_version_data = {
+                selector: {
+                    appid,
+                    secretName
+                },
+                fields: ['_id', '_rev'],
+            }
+            let secret_version_data_res = await DB.partitionedFind('secret_version_data', query_secret_version_data)
+            if (secret_version_data_res.docs.length > 0) {
+                for (const secret_version_data_item of secret_version_data_res.docs) {
+                    secret_version_data_item._deleted = true
+                }
+                await DB.bulk({ docs: secret_version_data_res.docs })
```

Comment:
if my understanding is correct, force deletion only deletes the secret metadata, but the secret versions are kept. 
Does it mean a secret in secret_version_data table will never be physically deleted from db?

Reply:
BTW, may I know where to find the usage of DB.destroy, DB.insert, and DB.bulk?